### PR TITLE
add support for windows on ARM64 native platform

### LINKED ISSUE #12144: [bug] ConanException: unsupported architecture armv8 on Windows on ARM
Attempting to use `conans.CMake.configure` on WIndows on ARM, fails with the error:

```
ConanException: unsupported architecture armv8
```

I tracked the error to be [here](https://github.com/conan-io/conan/blob/a620e5b7ba6b0d2128edd2056b7391b5fabcd77d/conans/client/tools/win.py#L447).

Apparently, Conan still isn't aware of the native ARM support that came with Visual Studio 2022 17.3 (there is now `vcvarsarm64.bat` which configures for native ARM development on the host ARM machine).

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 11 on ARM64
  * Compiler+version: Visual Studio 2022 17.3 Developer Preview for ARM64
  * Conan version: 1.52.0
  * Python version: 3.9.7

### PR DESCRIPTION
**Pull request by DoDoENT**
Changelog: Fix: Add support for windows on ARM64 native platform.
Docs: omit

Closes: #12144

### CODE REVIEW

File: conans/client/tools/win.py [hunk 1]
```diff
@@ -442,6 +442,11 @@ def vcvars_command(conanfile=None, arch=None, compiler_version=None, force=False
                        'x86_64': 'x86_amd64',
                        'armv7': 'x86_arm',
                        'armv8': 'x86_arm64'}.get(arch_setting)
+    elif arch_build == 'armv8':
```
Comment (memsharded) [inline conans/client/tools/win.py:445]: This is modifying only the legacy tool ``from conans``
This should be done as a secondary thing, as a backport, but this should be addressed first in ``from conan`` space.

Also some unittest would be necessary, please ask for guidance if you need it.
Thanks for contributing to Conan!

Comment (DoDoENT) [reply conans/client/tools/win.py:445]: Thank you for the pointers. I've added the same code also for non-legacy `from conan`.

I've also updated the only unit test I found that deals with testing the vcvars, but it also works with the legacy `from conans`. Also, how to run just this single test? I've run `python3 -m tox -e full` as described in the documentation and more than 1400 tests have failed, most of them completely unrelated to my change. Are they supposed to fail?

File: conans/test/functional/tools/old/vcvars/vcvars_arch_test.py [hunk 2]
```diff
@@ -53,6 +53,17 @@ def test_arch(self):
         settings.arch = 'x86'
         self.assert_vcvars_command(settings, "amd64_x86")
 
+        settings.arch_build = 'armv8'
+        settings.arch = 'armv8'
+
+        self.assert_vcvars_command(settings, "arm64")
+
+        self.arch = 'x86'
```
Comment (memsharded) [inline conans/test/functional/tools/old/vcvars/vcvars_arch_test.py]:
```suggestion
        settings.arch = 'x86'
```

File: conans/test/functional/tools/old/vcvars/vcvars_arch_test.py [hunk 3]
```diff
@@ -53,6 +53,17 @@ def test_arch(self):
...
+        self.assert_vcvars_command(settings, "arm64_x86")
+
+        self.arch = 'x86_64'
```
Comment (memsharded) [inline conans/test/functional/tools/old/vcvars/vcvars_arch_test.py]:
```suggestion
        settings.arch = 'x86_64'
```

**Pull request merged**